### DISCUSSION TITLE
Exclude actor from camera reset

### DISCUSSION THREAD

**Discussion by MattTheCuber** [pyvista/pyvista#5692] [Q&A] [answered]
Is there a way using PyVista or VTK to exclude an object from the scene camera resets? Basically, we have a textured floor object that expands far beyond the actors in the scene and the view automatically opens with the camera fitting this object. This object is also included in the scene bounding box visualization.

In MATLAB I can achieve this by using the `XLimInclude` and `YLimInclude` properties on a graphics object ([see here](https://undocumentedmatlab.com/articles/plot-liminclude-properties)).

**Comment by user27182** [comment]
There is a way to turn off the visibility of individual actors in a scene. However, the camera reset methods in pyvista do not currently take actor visibility into account. I believe this is a bug, see #5179. 

You can try calling the vtk method `ResetCamera()` as a workaround instead of calling pyvista's `reset_camera()`.

**Reply by MattTheCuber** [reply]
This is quite what we need for our case. We need to exclude a visible object from camera resets.

**Reply by user27182** [reply, marked as answer]
Right, so I think the workaround would look something like this:


``` python
import pyvista as pv
# create meshes for your scene
small_mesh = pv.Sphere(center = (0,0,1))
large_mesh = pv.Plane(i_size=100, j_size=100)

# Add regular meshes normally
pl = pv.Plotter()
pl.add_mesh(small_mesh)

# The large mesh should be added as an actor
large_mesh_actor = pv.Actor(mapper=pv.DataSetMapper(large_mesh))
large_mesh_actor.UseBoundsOff()  # key line to ignore the bounds of the large mesh
pl.add_actor(large_mesh_actor)

# use vtk methods directly to compute bounds of the scene
# This computation will take into consideration the `UseBoundsOff()` property
scene_bounds  = pl.renderer.ComputeVisiblePropBounds()

# Force camera reset with explicit bounds
pl.reset_camera(bounds=scene_bounds)

pl.show()
```

<img width="624" alt="image" src="https://github.com/pyvista/pyvista/assets/89109579/5ccbb88f-624d-4d3f-a585-5f57e9de84f8">

**Reply by MattTheCuber** [reply]
This is exactly what we needed. Thanks!

**Reply by MattTheCuber** [reply]
Just a note @user27182: we discovered that the `UseBoundsOff()` worked all on its own without the need for `scene_bounds`.

Additonally, the `UseBoundsOff()` seems to introduce a weird clipping issue (see video below). Any idea why this is?


https://github.com/pyvista/pyvista/assets/32849887/e541de81-dac4-4560-bbf7-289ec6406c8a

**Reply by MattTheCuber** [reply]
Also bounding box, ruler, and show edges all include the actor 😕 

![image](https://github.com/pyvista/pyvista/assets/32849887/265e94ba-b4c5-4076-8502-b62a90c5818c)

**Reply by user27182** [reply]
@MattTheCuber 
There is now `use_bounds` merged in #7206 along with the option to use `Plotter.compute_bounds` for some specialized bounds calcs. And some bounds issues were fixed in https://github.com/pyvista/pyvista/pull/7184

Feel free to try out the latest from `main` branch to see if these updates resolve your issue.

**Reply by MattTheCuber** [reply]
This works, thank you!
